docs/DeveloperGuide.md

```diff
@@ -257,329 +257,368 @@ _{Explain here how the data archiving feature will be implemented}_
 
-* 2a. The list is empty.
+  * 1a1. TAB displays a filtered list of students with the specified tag
   
-  Use case ends.
-
+    Use case ends.
+  
+* 1b. User requests for a list of students that match a specified name or address value
 
-* 3a. The given index is invalid.
+  * 1b1. TAB displays a list of students with matching name or address
 
-  * 3a1. TAB shows an error message. 
+    Use case ends.
 
-    Use case resumes at step 2.
-    
+<br/>
 
-* 3b. No data fields entered. 
-  * 3b1. TAB shows an error message. 
+#### Use Case: UC2 - Add a New Student
 
-    Use case resumes at step 3.
-    
+**MSS**
 
-* 3c. Content to be edited does not satisfy requirements.
+1. User requests to add a new student with specified fields
+2. TAB adds the student with the specified fields
 
-  * 3c1. TAB shows an error message with the requirements specified. 
-  * 3c2. User enters new data.
-  * Steps 3c1-3c2 are repeated until the new data filled in is satisfies the requirements.
-    
-    Use case resumes at step 4.
+   Use case ends.
 
-**Use case: Delete a person**
+**Extension**
 
-**MSS**
+* 1a. TAB detects that the entered student data does not satisfy requirements
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+    * 1a1. TAB displays an error message and the recommended changes
 
-    Use case ends.
+      Use case resumes at step 2.
 
-**Extensions**
+* 1b. TAB detects a duplicate student
 
-* 2a. The list is empty.
+    * 1b1. TAB displays an error message
 
-  Use case ends.
+      Use case ends.
 
-* 3a. The given index is invalid.
+<br/>
 
-    * 3a1. AddressBook shows an error message.
-      
-  Use case resumes at step 2.
+#### Use case: UC3 – Edit a Student
 
-**Use case: UC3 Track Student's Lesson Information**
+**MSS:**
 
-**MSS**
+1. User [<ins>gets a list of students (UC1)</ins>](Use-case:-UC1---Get-a-list-of-students)
+2. User requests to edit certain fields of specific student in the list
+3. TAB updates the corresponding fields of the student
 
-1.  User requests to list students
-2.  TAB shows a list of students
-3.  User requests to add a lesson for a specific student in the list
-4.  TAB adds the lesson to the student specified
-    
-    Use case ends.
+   Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
-  
+* 1a. The list is empty
+
   Use case ends.
 
-* 3a. The given index is invalid.
-  * 3a1. TAB shows an error message.
+* 2a. The given index is invalid
 
-* 3b. The specified end time is earlier than the start time
-  * 3b1. TAB shows an error message.
-    
-* 3c. There is an existing lesson for that student with the same lesson details.
-  * 3c1. TAB shows an error message. 
-    
-  Use case resumes at step 2.
+    * 2a1. TAB displays an error message
 
-**Use case: UC5 – Delete Student’s Lesson Information**
+      Use case resumes at step 1.
```

Comment:
IMO, the list is always there, so you get a list whether you like it or not. I understand that the user shouldn't request a list all over again (UC-1 step 1), but TAB will still display the list (UC-1 step 2) after a bad request. But because they are both part of the included use case (UC-1), I can only say that it starts again at step 1

Reply:
The way AB3 does it:
![image](https://user-images.githubusercontent.com/70138507/134840554-206a5d5a-3181-40c1-8133-af0e81156712.png)